### DISCUSSION TITLE
Horizontal scaling of the streaming API ?

### DISCUSSION THREAD

**Discussion by vladimir-stornest** [mastodon/mastodon#16492] [Q&A]
Hello everyone. I'm interested in scaling my Mastodon instance, including the Node.js service. I see in the documentation that I can increase the STREAMING_CLUSTER_NUM parameter for more worker processes, but what about horizontal scaling across multiple machines? 

As I know, Node.js with Websockets can be tricky for horizontal scaling (it's needed Pub/Sub, sticky sessions, or something else). Is that the case with the streaming API with Mastodon?

Thank you!

**Comment by Gargron** [comment]
You can spin up as many processes as you want and load balance from nginx. I see no obstacles to that at all. No sticky sessions necessary.

**Reply by vladimir-stornest** [reply]
Thanks for the quick response! This is my concern: 
Let's say we have a user Gargron who is connected with a WebSocket to a Nodejs process hosted on machine A, while user Vladimir is connected to machine B.
Then let's say Gargon needs to get a message (a live update), but the load balancer (Nginx) sends that specific request/message to machine B. Wondering if Gargon will get that message?

**Reply by weex** [reply]
Have you tried it? Seems to me you'd need something watching the db or redis for state change for connected users but I don't know what all is implemented.
